media/selectHandler.ts

```diff
@@ -22,81 +17,49 @@ export class SelectHandler {
-            } else {
-                this.singleSelect(parseInt((event.target as HTMLElement).getAttribute("data-offset")!));
-            }
-            (event.target as HTMLElement).focus();
-        }
-    }
-
     /**
      * @description Given an offset selects the elements. This does not clear the previously selected elements.
      * @param {number} offset Offset to select
+     * @param {boolean} force If force is not given, toggles selection. If force is true selects the element.
+     * If force is false deselects the element.
      */
-    private static selectOffset(offset: number): void {
+    private static toggleSelectOffset(offset: number, force?: boolean): void {
         const elements = getElementsWithGivenOffset(offset);
-        // We add the offset to the selection
-        const selectedOffsets = WebViewStateManager.getProperty("selected_offsets");
-        selectedOffsets.push(offset);
-        WebViewStateManager.setProperty("selected_offsets", selectedOffsets);
-        elements[0].classList.add("selected");
-        elements[1].classList.add("selected");
+        if (elements.length === 0) {
+            // Element may not be part of the DOM
+            return;
+        }
+        elements[0].classList.toggle("selected", force);
+        elements[1].classList.toggle("selected", force);
     }
 
-    /**
-     * @description Given an offset, selects the hex and ascii element
-     * @param {number} offset  The offset of the element you want to select
-     */
-    public static singleSelect(offset: number): void {
-        this.clearSelected();
-        this.selectOffset(offset);
-        const elements = getElementsWithGivenOffset(offset);
-        const byte_obj = retrieveSelectedByteObject(elements);
-        if (!byte_obj) return;
-        const littleEndian = (document.getElementById("endianness") as HTMLInputElement).value === "little";
-        populateDataInspector(byte_obj, littleEndian);
+    public getFocused(): number | undefined {
+        return this._focus;
     }
 
-    /**
-     * @description Selects the hex and ascii elements with the given offsets
-     * @param {number[]} offsets The list of offsets to select 
-     * @param {boolean} append Whether it should 
-     */
-    public static multiSelect(offsets: number[], append: boolean): void {
-        if (!append) this.clearSelected();
+    public setFocused(offset: number | undefined): void {
+        this._focus = offset;
+    }
+
+    public getSelected(): number[] {
+        return this._selection;
+    }
+
+    public setSelected(offsets: number[], forceRender = false): void {
+        const oldSelection = this._selection;
+
+        this._selection = [...offsets].sort((a: number, b: number) => a - b);
+        WebViewStateManager.setProperty("selected_offsets", this._selection);
+
+        const toRender = forceRender ? disjunction(oldSelection, this._selection) : relativeComplement(oldSelection, this._selection);
```

Comment:
A few more comments about what's going on here would be helpful as this is a little complex.